pkg/genericapiserver/genericapiserver.go

```diff
@@ -128,6 +131,18 @@ type APIGroupVersionOverride struct {
 	ResourceOverrides map[string]bool
 }
 
+// Info about a group version.
+type GroupVersionInfo struct {
+	GroupMeta latest.GroupMeta
+	// Info about the resources in this group version.
+	ResourcesStorageMap map[string]rest.Storage
+	// True, if this is the legacy core api ("/api").
+	IsLegacyCoreGroupVersion bool
+	// Optional: The group version to use for unversioned objects (like api.Status and unversioned.ListOptions).
+	// If nil, defaults to groupMeta.GroupVersion.
+	VersionForUnversionedObjects *unversioned.GroupVersion
```

Comment:
If we do #19018, OptionsExternalVersion gets removed.  The further we go
down that path, the less reason there is for Unversioned in the first
place.  For now, Keeping the name consistent is better.

On Tue, Dec 29, 2015 at 4:50 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/genericapiserver/genericapiserver.go
> <https://github.com/kubernetes/kubernetes/pull/19040#discussion_r48572624>
> :
>
> > @@ -128,6 +131,18 @@ type APIGroupVersionOverride struct {
> >  	ResourceOverrides map[string]bool
> >  }
> >
> > +// Info about a group version.
> > +type GroupVersionInfo struct {
> > +	GroupMeta latest.GroupMeta
> > +	// Info about the resources in this group version.
> > +	ResourcesStorageMap map[string]rest.Storage
> > +	// True, if this is the legacy core api ("/api").
> > +	IsLegacyCoreGroupVersion bool
> > +	// Optional: The group version to use for unversioned objects (like api.Status and unversioned.ListOptions).
> > +	// If nil, defaults to groupMeta.GroupVersion.
> > +	VersionForUnversionedObjects *unversioned.GroupVersion
>
> I didn't finished the review yet :) I was distracted here. I think we need
> to clean up the "OptionsExternalVersion". it's orthogonal to this PR , I'll
> open another issue.
>
> Two concrete comments here: 1. ListOptions is actually moved out of the
> unversioned package now, so the comment need to be updated. 2. can we
> change the variable name to OptionsExternalVersion to be consistent? I know
> that's a lame name (I invented that name btw :), because api.Status is not
> an option, but let's keep the name consistent so we can change all
> occurrences together when we come up with a better one. I don't like
> "VersionForUnversionedObjects" because 1. unversionedObejcts shouldn't have
> a version IMO, if it has, it's for backward compatibility 2. some use cases
> of this version is not on an unversioned object, the getOptions for
> example
> <https://github.com/kubernetes/kubernetes/blob/master/pkg/apiserver/api_installer.go#L276>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19040/files#r48572624>.
>